Street Trees/STConfirmationPageViewController.swift

```diff
@@ -0,0 +1,179 @@
+        {
+        case .TreeName:
+            return "Tree Information"
+        case .Address, .CityStateZip:
+            return "Location"
+        case .UserName, .Phone, .email:
+            return "Contact"
+        }
+    }
+}
+
+
+//**********************************************************************************************************************
+// MARK: - DataSource
+
+struct DataSource
+{
+    
+    private let dataSource: [[STConfirmationDetailRows]]
+    
+    var sections: Int {
+        return self.dataSource.count
+    }
+    
+    func itemsForSection(atIndex anIndex: Int) -> Int {
+        return self.dataSource[anIndex].count
+    }
+    
+    func item(forRow aRow: Int, inSection aSection: Int) -> STConfirmationDetailRows {
+        return self.dataSource[aSection][aRow]
+    }
+    
+    init(dataSource:[[STConfirmationDetailRows]]) {
+        self.dataSource = dataSource
+    }
+}
+
+//**********************************************************************************************************************
+// MARK: - Protocol
+
+protocol STConfirmationPageViewControllerDelegate: NSObjectProtocol
+{
+    func confirmationFormViewController(form: STConfirmationPageViewController, didCompleteWithWufooForm anWufooForm: STTKWufooForm)
+}
+
+//**********************************************************************************************************************
+
+class STConfirmationPageViewController: UIViewController, UITableViewDataSource, UITableViewDelegate, TreeDescription, Address, Contact
+{
+    
+    var treeDescription:STPKTreeDescription?
+    var address: STTKStreetAddress?
+    var contact: STTKContact?
+ 
+    
+    var dataSource: DataSource = DataSource(dataSource: [[.TreeName], [.Address, .CityStateZip], [.UserName, .Phone, .email]])
+    
+    weak var  delegate:STConfirmationPageViewControllerDelegate?
+
+    // MARK: - Table View Methods:
+    
+    func numberOfSectionsInTableView(tableView: UITableView) -> Int
+    {
+        return dataSource.sections
+    }
+    
+    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int
+    {
+        return dataSource.itemsForSection(atIndex: section)
+    }
+    
+    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell
+    {
+        let cell = tableView.dequeueReusableCellWithIdentifier("emptyCell", forIndexPath: indexPath)
+
+        let kindOfRow = dataSource.item(forRow: indexPath.row, inSection: indexPath.section)
+        
+        var textLabel:String? = nil
+        var imageView:UIImage? = nil
+        
+        switch kindOfRow
+        {
+        case .TreeName:
+            textLabel = treeDescription?.name ?? "Error retrieving tree name"
+            imageView = treeDescription?.image() ?? UIImage(named: "blankMap")
+        case .Address:
+            textLabel = (address?.streetAddress)!  + " " +  (address?.secondaryAddress)! ?? "Error retrieving address"
```

Comment:
Forcing unwraps will causes crashes and make the ?? useless.

Properties don't require () around them when you are string to concatenate them into a string.
address?.streetAddress!  + " " +  address?.secondaryAddress!
or
"\(address?.streetAddress!)  \(address?.secondaryAddress!)"

Better yet:
let address = address?.streetAddress ?? "No address specified"
let address2 = address?.secondaryAddress ?? "" // 2nd address not required so convert nil to an empty string.
textLabel = "\(address) \(address2)"